### PR TITLE
Validation jobs SQL AAD + net462 + packageref

### PR DESCRIPTION
**Pull request by chenriksson**
Main parts to review for SQL AAD:
- [src/NuGet.Services.Validation.Orchestrator/Job.cs](https://github.com/NuGet/NuGet.Jobs/pull/464/files#diff-aec90de495b800c4fb9a385ba99be23f)
- [src/Validation.Common.Job/JsonConfigurationJob.cs](https://github.com/NuGet/NuGet.Jobs/pull/464/files#diff-47a9b7b6c76746cff4bdd13928e95d43)
- [src/Validation.PackageSigning.ProcessSignature/Job.cs](https://github.com/NuGet/NuGet.Jobs/pull/464/files#diff-88e67d5a6941e4d1d30f8a9364867263)

### CODE REVIEW

File: src/NuGet.Services.Validation.Orchestrator/Job.cs [hunk 1]
```diff
@@ -78,9 +80,14 @@ public class Job : JobBase
 
         public override void Init(IServiceContainer serviceContainer, IDictionary<string, string> jobArgsDictionary)
         {
+            ISecretInjector secretInjector;
+
             var configurationFilename = JobConfigurationManager.GetArgument(jobArgsDictionary, ConfigurationArgument);
             _validateOnly = JobConfigurationManager.TryGetBoolArgument(jobArgsDictionary, ValidateArgument, defaultValue: false);
-            _serviceProvider = GetServiceProvider(GetConfigurationRoot(configurationFilename, _validateOnly));
+
+            var configurationRoot = GetConfigurationRoot(configurationFilename, _validateOnly, out secretInjector);
```
Comment (scottbommarito) [inline src/NuGet.Services.Validation.Orchestrator/Job.cs]: nit: `out var secretInjector` so you don't have to declare it above?

File: src/CopyAzureContainer/CopyAzureContainer.csproj [hunk 2]
```diff
@@ -86,7 +58,20 @@
...
+      <Version>1.0.0</Version>
+    </PackageReference>
+    <PackageReference Include="System.ComponentModel.EventBasedAsync">
+      <Version>4.0.11</Version>
+    </PackageReference>
+    <PackageReference Include="System.Net.Http">
+      <Version>4.3.3</Version>
+    </PackageReference>
+    <PackageReference Include="WindowsAzure.Storage">
+      <Version>7.1.2</Version>
```
Comment (loic-sharma) [inline src/CopyAzureContainer/CopyAzureContainer.csproj]: >7.1.2 [](start = 15, length = 5)

Previous version was `8.1.1`, is this downgrade intentional?

File: src/NuGet.Jobs.Common/NuGet.Jobs.Common.csproj [hunk 3]
```diff
@@ -222,32 +65,32 @@
...
   </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="Dapper.StrongName">
+      <Version>1.50.2</Version>
+    </PackageReference>
+    <PackageReference Include="NuGet.Services.Configuration">
+      <Version>2.25.0</Version>
+    </PackageReference>
+    <PackageReference Include="NuGet.Services.KeyVault">
+      <Version>2.26.0-master-33196</Version>
```
Comment (loic-sharma) [inline src/NuGet.Jobs.Common/NuGet.Jobs.Common.csproj]: >2.26.0-master-33196 [](start = 15, length = 19)

Is it possible to keep all these versions consistent?

File: src/SnapshotAzureBlob/SnapshotAzureBlob.csproj [hunk 4]
```diff
@@ -74,13 +46,23 @@
...
   </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="Microsoft.Extensions.Logging.Abstractions">
+      <Version>1.0.0</Version>
+    </PackageReference>
+    <PackageReference Include="System.ComponentModel.EventBasedAsync">
+      <Version>4.0.11</Version>
+    </PackageReference>
+    <PackageReference Include="WindowsAzure.Storage">
+      <Version>7.1.2</Version>
```
Comment (loic-sharma) [inline src/SnapshotAzureBlob/SnapshotAzureBlob.csproj]: >7.1.2 [](start = 15, length = 5)

Same thing here, this used to be version 8.1.1

File: src/CopyAzureContainer/CopyAzureContainer.csproj (hunk 2 above)
Comment (chenriksson) [reply src/CopyAzureContainer/CopyAzureContainer.csproj]: Yes - trying to clean up dependencies across our projects while I'm touching everything. Most of our projects depend on WindowsAzure.Store, v7.1.2 so I'm aligning to that.

Not absolutely necessary in this case, but in some cases I aligned to specific dependency versions to remove build warnings. Also, migration to PackageRef allowed me to remove some indirect dependencies which were explicitly referenced.

**Pull request merged**